pkg/antctl/raw/multicluster/join.go

```diff
@@ -0,0 +1,288 @@
+		raw, err := ioutil.ReadFile(o.TokenSecretFile)
+		if err != nil {
+			return err
+		}
+		o.Secret, err = unmarshallSecret(raw, false)
+		if err != nil {
+			return fmt.Errorf("failed to unmarshall Secret from token Secret file: %v", err)
+		}
+		o.Secret.Namespace = o.Namespace
+	}
+
+	if o.LeaderClusterID == "" {
+		return fmt.Errorf("the clusterid of leader cluster is required")
+	}
+	if o.LeaderAPIServer == "" {
+		return fmt.Errorf("the API server of the leader cluster is required")
+	}
+	if o.TokenSecretName == "" && o.Secret == nil {
+		return fmt.Errorf("specify at least one of the leader cluster Secret and the Secret in the config file")
+	}
+	if o.Namespace == "" {
+		return fmt.Errorf("the the leader cluster Namespace is required")
+	}
+	if o.ClusterSetID == "" {
+		return fmt.Errorf("the ClusterSet ID is required")
+	}
+	if o.ClusterID == "" {
+		return fmt.Errorf("the member clusterID is required")
+	}
+	if o.Namespace == "" {
+		return fmt.Errorf("the member cluster Namespace cannot is required")
+	}
+
+	return nil
+}
+
+func yamlUnmarshall(raw []byte, v interface{}) error {
+	d := yaml.NewDecoder(bytes.NewReader(raw))
+	d.KnownFields(false)
+	return d.Decode(v)
+}
+
+func unmarshallSecret(raw []byte, configFile bool) (*v1.Secret, error) {
+	decoder := yamlutil.NewYAMLOrJSONDecoder(bytes.NewReader(raw), 100)
+	secret := &v1.Secret{}
+	if configFile {
+		// The config file is a YAML file, so we need to decode it to a Secret.
+		// And the first object in the file is the ClusterSetJoinConfig, so we need to skip it.
+		u := unstructured.Unstructured{}
+		if err := decoder.Decode(&u); err != nil {
+			return nil, err
+		}
+	}
+	if err := decoder.Decode(secret); err != nil {
+		return nil, err
+	}
+
+	return secret, nil
+}
+
+var joinExamples = strings.Trim(`
+# Join the ClusterSet from a member cluster with command line options.
+  $ antctl mc join --leader-namespace=antrea-multicluster \
+				   --leader-clusterid=cluster-north \
+				   --secret=member-access-token \
+				   --leader-apiserver=https://172.18.0.3:6443 \
+				   --clusterset=clusterset \
+				   --namespace=kube-system \
+				   --clusterid=cluster-east
+
+# Join the ClusterSet from a member cluster with a token Secret file.
+  $ antctl mc join --leader-namespace=antrea-multicluster \
+				   --leader-clusterid=cluster-north \
+				   --leader-apiserver=https://172.18.0.3:6443 \
+				   --clusterset=clusterset \
+				   --namespace=kube-system \
+				   --clusterid=cluster-east \
+				   --token-secret-file=member-access-token.yaml
+
+# Join the ClusterSet from a member cluster with a config file. 
+  $ antctl mc join --config-file ./config.yaml
+
+# Config file example:
+apiVersion: multicluster.antrea.io/v1alpha1
+kind: ClusterSetJoinConfig
+namespace: kube-system
+clusterID: cluster-east
+leaderNamespace: antrea-multicluster
+tokenSecretName: member-access-token
+leaderAPIServer: https://172.18.0.3:6443
+leaderClusterID: cluster-north
+clusterSetID: clusterset
```

Comment:
Probably we should add a Secret to the example, like:

```
---
apiVersion: v1
kind: Secret
metadata:
  name: default-member-token
type: Opaque
data:
  ... # Generated by "init" or "create membertoken" command
```

Reply:
An secret example added.